### ISSUE TITLE
Docker variables for database are not set

### ISSUE THREAD

**Issue by sumdog** [paperwork/paperwork#696]
With the current docker-compose script, the database migrations within the `docker-runner.sh` cannot run since the database configuration hasn't been established yet. I was planning on submitting a pull that fixes this. I attempted to write out the configuration based on the environment injected from the linked database container: 

```
cat << EOF > /app/app/storage/config/database.json
{"driver":"mysql",
 "database":"$DB_ENV_MYSQL_DATABASE",
 "host":"$DB_PORT_3306_TCP_ADDR",
 "username":"$DB_ENV_MYSQL_USER",
 "password":"$DB_ENV_MYSQL_PASSWORD",
 "port":"$DB_PORT_3306_TCP_PORT"
}
EOF

# Ensure migration is performed and database is not empty.
if [[ ! -e migrated ]]; then
    php artisan migrate

    touch migrated
fi

/run.sh
```

After this, when I start paperwork, I get to the 2nd screen "Customize your Paperwork" but when I attempt to move past it, I get the following:

```
Configuration has not been set correctly. Please try again. 
```

I've been trying to understand the setup code but am a bit lost with the controller structure. What other configuration file(s) do I need to write in order to automate the DB creation process with docker compose?

**Comment by Liongold** [comment]
Are you using develop branch?

**Comment by sumdog** [comment]
No, I was on master. I checked out the master branch and tried it. This is my current `docker-runner.sh`:

```
cat << EOF > /app/app/storage/config/database.json
{"driver":"mysql",
 "database":"$DB_ENV_MYSQL_DATABASE",
 "host":"$DB_PORT_3306_TCP_ADDR",
 "username":"$DB_ENV_MYSQL_USER",
 "password":"$DB_ENV_MYSQL_PASSWORD",
 "port":"$DB_PORT_3306_TCP_PORT"
}
EOF

# Ensure migration is performed and database is not empty.
if [[ ! -e migrated ]]; then
    # ensure MySQL docker container is up
    while ! mysqladmin ping -h "$DB_PORT_3306_TCP_ADDR" --silent; do
      echo "Waiting for MySQL container..."
      sleep 1
    done
    php artisan migrate
    touch migrated
fi

cat << EOF > /tmp/a #/app/app/storage/config/paperwork.json
{"registration":"$PR_ALLOW_REGISTRATION",
 "forgot_password":"$PR_ALLOW_PASSWORD_RESET",
 "userAgentLanguage":"$PR_USER_AGENT_LANG",
 "showIssueReportingLink":"$PR_SHOW_REPORT_LINK"}
EOF

echo "4" > /app/app/storage/config/setup

/run.sh
```

The `paperwork.json` variables are in the `Dockerfile`. I was trying to avoid having to configure that part, but I couldn't get past the setup screen on the mater branch without it. However if I do this on develop, the database migrations fail:

```
web_1  |   [Illuminate\Database\QueryException]                                                                                                                                                                                                                                                                                      
web_1  |   SQLSTATE[42000]: Syntax error or access violation: 1064 You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'CURRENT_TIMESTAMP' at line 1 (SQL: ALTER TABLE attachment_version ALTER COLUMN created_at SET DEFAULT CURRENT_TIMESTAMP;)  
```

**Comment by sumdog** [comment]
It looks like that SQL statement is coming from the `FixTimestampsForPostgres` class in `2016_10_03_203511_fix_timestamps_for_postgres.php`, but I'm running MySQL. That's probably a bug? Does the migration system your using need a flag to only apply to certain database types? 

If I change setup to 8 and write out the database and paperwork configs on develop, I get to the main page. Tried it on mater and that worked as well. I've created a pull request with these changes for master:

https://github.com/twostairs/paperwork/pull/698
